libwebauthn/src/management/credential_management.rs

```diff
@@ -0,0 +1,319 @@
+use crate::{
+    ops::webauthn::UserVerificationRequirement,
+    pin::{PinProvider, PinUvAuthProtocol},
+    proto::ctap2::{
+        Ctap2, Ctap2AuthTokenPermissionRole, Ctap2CredentialData,
+        Ctap2CredentialManagementMetadata, Ctap2CredentialManagementRequest, Ctap2GetInfoResponse,
+        Ctap2PublicKeyCredentialDescriptor, Ctap2PublicKeyCredentialRpEntity,
+        Ctap2PublicKeyCredentialUserEntity, Ctap2UserVerifiableRequest,
+    },
+    transport::{
+        error::{CtapError, Error},
+        Channel,
+    },
+    webauthn::{handle_errors, user_verification, UsedPinUvAuthToken},
+};
+use async_trait::async_trait;
+use serde_bytes::ByteBuf;
+use serde_cbor::ser::to_vec;
+use std::time::Duration;
+use tracing::info;
+
+#[async_trait]
+pub trait CredentialManagement {
+    async fn get_credential_metadata(
+        &mut self,
+        pin_provider: &mut Box<dyn PinProvider>,
+        timeout: Duration,
+    ) -> Result<Ctap2CredentialManagementMetadata, Error>;
+    async fn enumerate_rps_begin(
```

Comment:
Would it make sense for the higher level APIs (`enumerate_rps`, `enumerate_credentials`) to return `Iterator`s?

The only downside I can think of is that we couldn't easily export the `count`.

No need to address this in this PR.